### PR TITLE
Fixes: Delete chat button should follow designs

### LINKED ISSUE #1603: Delete chat button should follow designs
The delete button of the Delete Chat Popup has a weird style mix of a standard/primary button and a warn button.

Screenshot 2021-01-11 at 16 54 22

### PR DESCRIPTION
**Pull request by staked-smart-ace**
Fixes #1603

### CODE REVIEW

File: ui/imports/Themes/LightTheme.qml [hunk 1]
```diff
@@ -55,6 +55,7 @@ Theme {
     property color buttonSecondaryColor: darkGrey
     property color buttonDisabledForegroundColor: buttonSecondaryColor
     property color buttonDisabledBackgroundColor: grey
+    property color buttonWarnBackgroundColor: "#FFEAEE"
```
Comment (jrainville) [inline ui/imports/Themes/LightTheme.qml:58]: You'll need to add this color to the dark theme as well. For now, you can use the same hex color

**Review by jrainville — COMMENTED** [review]
Great idea to add the `warn` type. 

One thing we could add to it is that when the button type is `warn`, we change the text color to `Style.current.danger` (red) that way we don't have to pass the red color ourselves.

That way, the ConfirmationDialog wouldn't need to set ` color: Style.current.danger`

File: ui/imports/Themes/LightTheme.qml (hunk 1 above)
Comment (staked-smart-ace) [reply ui/imports/Themes/LightTheme.qml:58]: ok, I see

**Review by PascalPrecht — CHANGES_REQUESTED** [review]
Nice work @staked-smart-ace !
In addition to @jrainville 's comment, we need to make sure the hover effect of the new `warn` type is behaving correctly as well.

This is not the case right now. Notice how the button turns to a "normal" one when hovered.

**Comment by staked-smart-ace** [comment]
> In addition to @jrainville 's comment, we need to make sure the hover effect of the new `warn` type is behaving correctly as well.
> 
> This is not the case right now. Notice how the button turns to a "normal" one when hovered.
@jrainville @PascalPrecht 
I didn't find the Figma design for the warn type of the button.
If you provide the correct design, it would be great for me to fix it quickly. :)

**Comment by PascalPrecht** [comment]
@staked-smart-ace that's a good point. I think the hover style is just like the base warning one, just that the background color needs to be darker. IIRC there's already a `Qt.darker()` in place for that

**Comment by staked-smart-ace** [comment]
I just added hover status to warn StatusButton.

**Review by jrainville — COMMENTED** [review]
I tested and it looks good. One small thing missing would be my previous suggestion that if the `type` is `warn`, it should default the `color` to `danger` (red).é That way, you don't have to pass `red` to the buttons directly in the Dialog

**Comment by staked-smart-ace** [comment]
I just updated as you mentioned. :)

File: ui/shared/status/StatusButton.qml [hunk 2]
```diff
@@ -69,7 +69,9 @@ Button {
             anchors.left: iconLoader.active ? iconLoader.right : parent.left
             anchors.leftMargin: iconLoader.active ? Style.current.smallPadding : 0
             color: !enabled ? Style.current.buttonDisabledForegroundColor : 
-              (hovered || highlighted) ? Style.current.blue : control.color
+              (hovered || highlighted) ?
+                type === "warn" ? Style.current.danger : Style.current.blue :
```
Comment (jrainville) [inline ui/shared/status/StatusButton.qml:73]: Instead fo changing it here, you could leave it as you had before (`control.color`), and just change the actual color above like so:
```qml
property color color: type === "warn" ? Style.current.danger : Style.current.buttonForegroundColor
```

That way, the color stays configurable even if the type is warn

**Comment by staked-smart-ace** [comment]
I just updated

File: ui/shared/status/StatusButton.qml [hunk 3]
```diff
@@ -68,8 +68,7 @@ Button {
             anchors.right: iconLoader.active ? undefined : parent.right
             anchors.left: iconLoader.active ? iconLoader.right : parent.left
             anchors.leftMargin: iconLoader.active ? Style.current.smallPadding : 0
-            color: !enabled ? Style.current.buttonDisabledForegroundColor : 
-              (hovered || highlighted) ? Style.current.blue : control.color
+            color: !enabled ? Style.current.buttonDisabledForegroundColor : control.color
```
Comment (jrainville) [inline ui/shared/status/StatusButton.qml:71]: I think you still need this part ` (hovered || highlighted) ? Style.current.blue : control.color`

You probably need to add a check for `warn` too

```
 (type !== "warn" && (hovered || highlighted)) ? Style.current.blue : control.color
 ```

Comment (staked-smart-ace) [reply ui/shared/status/StatusButton.qml]: Currently, it is working well, however, let me follow you ;)

**Comment by iurimatias** [comment]
@staked-smart-ace could you rebase this PR please

**Comment by staked-smart-ace** [comment]
ok, I see

**Comment by staked-smart-ace** [comment]
> @staked-smart-ace could you rebase this PR please

@iurimatias Could your merge it PR if it looks like?

**Comment by PascalPrecht** [comment]
This has landed as https://github.com/status-im/nim-status-client/commit/086c868bdc5733c989eb9cb9e1c0424902055731

@staked-smart-ace I squashed your commits and rebased them on top of latest `master`

**Pull request closed without merging**